### PR TITLE
[BOLT] Fix counts aggregation in merge-fdata

### PR DESCRIPTION
**Pull request by aaupov**
merge-fdata used to consider misprediction count as part of "signature",
or the aggregation key. This prevented it from collapsing profile lines
with different misprediction counts, which resulted in duplicate
`(from, to)` pairs with different misprediction and execution counts.

Fix that by splitting out misprediction count and accumulating it
separately.

Test Plan: updated bolt/test/merge-fdata-lbr-mode.test

### CODE REVIEW

**Review by maksfb — COMMENTED** [review]
Since you are switching to regex, are there any changes in parsing performance?

File: bolt/tools/merge-fdata/merge-fdata.cpp [hunk 1]
```diff
@@ -330,14 +347,14 @@ void mergeLegacyProfiles(const SmallVectorImpl<std::string> &Filenames) {
   ProfileTy MergedProfile;
   for (const auto &[Thread, Profile] : ParsedProfiles)
     for (const auto &[Key, Value] : Profile) {
-      uint64_t Count = MergedProfile.lookup(Key) + Value;
+      auto Count = MergedProfile.lookup(Key) + Value;
```
Comment (ayermolo) [inline bolt/tools/merge-fdata/merge-fdata.cpp:350]: Why change to auto?

**Comment by aaupov** [comment]
> Since you are switching to regex, are there any changes in parsing performance?

With updated version (on top of #119942), it's neutral/slightly faster than current merge-fdata 
> 1.04 ± 0.07 times faster

File: bolt/tools/merge-fdata/merge-fdata.cpp (hunk 1 above)
Comment (aaupov) [reply bolt/tools/merge-fdata/merge-fdata.cpp]: The type of `Count` is changed, made explicit in updated version

**Review by maksfb — APPROVED** [review]
LGTM

**Pull request merged**